cloudbuild-task-azp/src/Tool.ts

```diff
@@ -1,19 +1,36 @@
 import * as task from 'azure-pipelines-task-lib/task';
 import * as contracts from '@aarnott/cloudbuild-task-contracts';
-import { Stream } from 'stream';
 
 export class Tool implements contracts.Tool {
-	spawn(path: string, args?: string[], options?: contracts.ToolOptions): Promise<number> {
-		const p = task.tool(path);
-		if (args) {
-			p.arg(args);
+	spawn(commandLine: string, args?: string[] | string, options?: contracts.ToolOptions): Promise<number> {
+		// If the args were passed in as a flat string,
+		// move them to the commandLine because they've already been properly quoted and escaped.
+		// The GitHub exec function will manage splitting them into an args array.
+		if (typeof args === 'string') {
+			commandLine += ' ' + args;
+			args = undefined;
 		}
 
+		const commandLineAsArgs = contracts.Utilities.parseArguments(commandLine);
+		if (commandLineAsArgs.length === 0) {
+			throw new Error(`Parameter 'commandLine' cannot be null or empty.`)
+		}
+
+		const p = task.tool(commandLineAsArgs[0]);
```

Comment:
Yup. A few points changed my mind:
1. Both github actions and AzP have the odd behavior of writing the command line to the stdout given by the caller. So they're self-consistent.
1. Given the abstraction is supposed to emulate the cloud behavior as closely as possible, that includes consistent behavior even if I disagree with it.
1. Callers already can use `child_process.spawn` directly to avoid the stdout pollution, so I didn't feel the need to introduce another alternative. Two options that cover all the use cases seemed like enough that I didn't need to add a 3rd one that was redundant.

That said, I'm rather fond of my API's flexibility to take args as a flat string, included or not as part of the command itself. It's a little better than GitHub's API, moderately better than AzP's, and heaps better than node's. :) But the most critical part of the functionality _is_ exposed as a public utility method so folks using `child_process` directly can still tap into it.